src/modules/CurrencyModule.cpp

```diff
@@ -902,7 +902,219 @@ CurrencyModule::CurrencyModule(UmikoBot* client) : Module("currency", true), m_c
 
 	});
+		RegisterCommand(Commands::CURRENCY_BRIBE, "bribe", [this](Discord::Client& client, const Discord::Message& message, const Discord::Channel& channel)
+		{
+
+			QStringList args = message.content().split(' ');
+			auto& config = getServerData(channel.guildId());
+			snowflake_t authorId = message.author().id();
+
+			int remainingJailTime = guildList[channel.guildId()][getUserIndex(channel.guildId(), authorId)].jailTimer->remainingTime();
+			bool inJail = guildList[channel.guildId()][getUserIndex(channel.guildId(), authorId)].jailTimer->isActive();
+			auto& authorCurrency = guildList[channel.guildId()][getUserIndex(channel.guildId(), authorId)];
+
+
+			if (inJail == false)
+			{
+				QString output = QString(
+					":police_officer: **Hey you're not in JAIL!** :police_officer:\n"
+				);
+				client.createMessage(message.channelId(), output);
+				return;
+			}
+
+			if (args.size() != 2)
+			{
+				client.createMessage(message.channelId(), "**Wrong Usage of Command!**");
+				return;
+			}
+
+			QRegExp re("[+]?\\d*\\.?\\d+");
+			if (!re.exactMatch(args.at(1)))
+			{
+				client.createMessage(message.channelId(), "**Your bribe amount must be valid!**");
+				return;
+			}
+
+			double amountToBribe = args.at(1).toDouble();
+
+			auto guildID = channel.guildId();
+			QObject::connect(authorCurrency.jailTimer, &QTimer::timeout, [this, &client, guildID, authorId]()
+				{
+					guildList[guildID][getUserIndex(guildID, authorId)].isBribeUsed = false;
+				});
+
+			if (inJail && authorCurrency.isBribeUsed)
+			{
+				QString output = QString(
+					":police_officer: **You already tried to bribe me and failed... Do you want me to extend your sentence again?** :police_officer:\n"
+				);
 
+				client.createMessage(message.channelId(), output);
+				return;
+			}
+
+			if (inJail && amountToBribe > config.bribeMaxAmount)
+			{
+				QString maxBribeAmount = QString::number(config.bribeMaxAmount);
+				QString output = QString(
+					":police_officer: **If I take more than `%1` %2 then I might get caught... ** :police_officer:\n"
+				).arg(maxBribeAmount, config.currencySymbol);
+
+				client.createMessage(message.channelId(), output);
+				return;
+			}
+
+			if (inJail && amountToBribe < config.bribeLeastAmount)
+			{
+				QString leastBribeAmount = QString::number(config.bribeLeastAmount);
+				QString output = QString(
+					":police_officer: **Pfft! Such measly amounts... Do you want to be in jail for longer?** :police_officer:\n"
+					"*(You can always give me `%1 %2` or more... maybe then I could do something?)*\n"
+				).arg(leastBribeAmount, config.currencySymbol);
+
+				client.createMessage(message.channelId(), output);
+				return;
+			}
+
+			if (inJail && authorCurrency.currency() - (amountToBribe) < debtMax)
+			{
+				client.createMessage(message.channelId(), ":police_officer: **I would love to be bribed with that much money but unfortunately you can't afford it.** :police_officer:");
+				return;
+			}
```

Comment:
ARen't we guaranteed to be in jail if we reach this code? I also think you should use `else if`s where possible so we don't run into dual cases

Reply:
Yeah, Okay I'm fixing this on next commit 😄

Reply:
Done!